### PR TITLE
feat: full path and method for macro introspection metadata

### PR DESCRIPTION
**Pull request by hunterwilhelm**
**Goal**: Enable macros to introspect route metadata (path and method) during route registration for documentation generation and analysis.

**Features Added**:

1. **Route metadata in macro introspection**
   - Added `MacroIntrospectionMetadata` with `path` and `method`
   - `introspect` now receives route metadata as a second parameter

2. **Path resolution**
   - Paths are normalized before introspection
   - Prefixes are resolved so metadata uses the final route path
   - Groups defer introspection until routes are added to the parent with fully resolved paths

3. **Guard-level macro introspection**
   - Guard-level macros are introspected per route with the resolved path
   - Introspection runs once per route with complete metadata

**Behavior Changes**:

- `introspect` receives `(introspectOptions, metadata)` where `metadata` contains `{ path: string, method: HTTPMethod }`
- Introspection runs only when route metadata is available
- Paths include prefixes and group paths (e.g., `/prefix/group/route/:id`)
- Works with nested groups, guards, and route combinations

**Use Case**: Enables macros to generate documentation, analyze routes, or perform static analysis using route path and method information.

## Summary by CodeRabbit

* **New Features**
  * Per-route macro introspection now receives route context (path and HTTP method) when applicable.
  * Introspection is executed per-route (only when full route metadata is available) and integrates across guards/groups without changing non-macro behavior.
  * MacroIntrospectionMetadata type is exposed in the public API.

* **Tests**
  * Expanded tests covering macro introspection across guards, groups, nested routes, and edge cases.

### CODE REVIEW

**Comment by hunterwilhelm** [comment]
Fixes https://github.com/elysiajs/elysia/issues/1498

File: src/index.ts [hunk 1]
```diff
@@ -487,7 +498,51 @@ export default class Elysia<
...
+					introspectFn(introspectOptions, { path, method })
+				}
+			}
+		}
+
+		// Skip macro introspection for routes added to child instances inside groups
+		// Macro introspection will happen when routes are added to the parent with the fully-resolved path
+		if (!this.skipMacroIntrospection) {
+			this.applyMacro(localHook, localHook, { metadata: { path, method } })
+		}
```
[the remark this answers is not in the document]
Comment (hunterwilhelm) [reply src/index.ts:559]: Added a test for this and resolved this issue.

**Comment by hunterwilhelm** [comment]
@SaltyAom I was able to update index.ts so that it now passes all of the new tests for this feature. However, I ended up making more changes than I would have preferred, including adding a few instance variables that I’m not fully convinced are necessary.

If there’s a more elegant or idiomatic way to approach this within Elysia’s design, I’d be happy to revise the PR.

**Comment by SaltyAom** [comment]
I'll try to come up with something easier

For a confirmation, all you would like to be added is path, and method right?

**Comment by hunterwilhelm** [comment]
Correct, as long as it's the fullly resolved path

**Comment by Necrelox** [comment]
any feedback on this pr ? 🥺
